replica/tablets.cc

```diff
@@ -641,6 +639,9 @@ tablet_sstable_set::create_single_key_sstable_reader(
         const sstables::sstable_predicate& predicate) const {
     // The singular partition_range start bound must be engaged.
     auto idx = group_of(pr.start()->value().token());
+    if (!_sstable_sets.contains(idx)) {
+        return make_empty_flat_reader_v2(std::move(schema), std::move(permit));
+    }
     const auto& set = find_sstable_set(idx);
```

Comment:
why do we need this change? left over from previous version? we need to throw if set is not found on single key reads.